### PR TITLE
Added to bid function

### PR DESCRIPTION
**Pull request by Leman-Y**

### CODE REVIEW

File: contracts/Auction.sol [hunk 1]
```diff
@@ -96,16 +97,42 @@ contract Auction {
...
+            msg.value (uint): number of wei sent with the message
+        */
+        bid_history.push(bid_record(msg.sender, msg.value, block.timestamp));
+        
+        if (current_highest_bid == 0){
+            current_highest_bid = msg.value;
+            current_highest_bidder = msg.sender; 
+        }
+        else{
+            current_highest_bidder.transfer(address(this).balance);
```
Comment (xiaoxiazhang00) [inline contracts/Auction.sol:150]: could you explain a bit this line, what's `address(this).balance` refer to, thanks !

File: contracts/Auction.sol [hunk 2]
```diff
@@ -96,16 +97,42 @@ contract Auction {
...
 
     /*
         Place a bid on the item. The bid is added to the history of bids list. Incoming bid must be higher than the highest bid otherwise reject the bid. If incoming bid
         is higher than highest bid, then the highest bidder's funds should be released back to he/she and update highest bid.
     */
-    function bid(address bidder, uint bid_amount) payable public {
-        bid_history.push(bid_record(bidder, bid_amount, block.timestamp));
+    function bid() payable public {
+        require(
+            block.timestamp <= end_time,
```
Comment (xiaoxiazhang00) [inline contracts/Auction.sol:129]: shall we also add the condition check for `ended` variable

File: contracts/Auction.sol (hunk 1 above)
Comment (Leman-Y) [reply contracts/Auction.sol:150]: if you googled it then you would know what it does. just playing haha. it refers to the smart contract's ethereum balance

File: contracts/Auction.sol (hunk 2 above)
Comment (Leman-Y) [reply contracts/Auction.sol:129]: uh maybe not sure yet. we might not even need the ended variable since we have the end_time

Comment (xiaoxiazhang00) [reply contracts/Auction.sol:129]: what if the owner close the auction, do we need another variable to handle this case ?

**Review by muskankapoor — APPROVED** [review]

**Pull request merged**